### PR TITLE
New rubocop

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: src/include/packager/repositories_include.rb [hunk 1]
```diff
@@ -483,11 +483,12 @@ def confirm_plain_repo
     # Check and install the packages needed for accessing the URL scheme.
     # @param scheme [String] URL scheme of the new repository
     def install_mount_package(scheme)
-      if scheme == "smb" && !File.exist?("/sbin/mount.cifs")
-        log.info("Installing missing 'cifs-mount' package...")
-        # install cifs-mount package
-        PackageSystem.CheckAndInstallPackages(["cifs-mount"])
-      end
+      return if scheme != "smb"
```
Comment (lslezak) [inline src/include/packager/repositories_include.rb]: NP: Just merging the lines?

```ruby
return if scheme != "smb" || File.exist?("/sbin/mount.cifs")
```

File: src/lib/packager/clients/repositories.rb [hunk 2]
```diff
@@ -187,9 +186,7 @@ def PriorityToString(priority)
         rest = Ops.subtract(rest, 1)
       end
 
-      if priority == @default_priority
-        ret = Ops.add(Ops.add(Ops.add(ret, " ("), _("Default")), ")")
-      end
+      ret = Ops.add(Ops.add(Ops.add(ret, " ("), _("Default")), ")") if priority == @default_priority
```
Comment (lslezak) [inline src/lib/packager/clients/repositories.rb]:
```suggestion
      ret +=  " (" + _("Default") + ")" if priority == @default_priority
```

File: src/lib/packager/clients/repositories.rb [hunk 3]
```diff
@@ -1252,9 +1248,7 @@ def SummaryDialog
                 end
                 schema = Builtins.tolower(Builtins.substring(url2, 0, 3))
 
-                if schema != "cd:" && schema != "dvd"
-                  to_refresh = Ops.add(to_refresh, 1)
-                end
+                to_refresh = Ops.add(to_refresh, 1) if schema != "cd:" && schema != "dvd"
```
Comment (lslezak) [inline src/lib/packager/clients/repositories.rb]:
```suggestion
                to_refresh += 1 if schema != "cd:" && schema != "dvd"
```

File: src/lib/y2packager/clients/inst_productsources.rb [hunk 4]
```diff
@@ -753,15 +745,11 @@ def ReadDialog
       @language_long = GetCurrentLang() if !Stage.initial
 
       # fallback if no LANG variable set
-      if @language_long.nil? || @language_long == ""
-        @language_long = Language.language
-      end
+      @language_long = Language.language if @language_long.nil? || @language_long == ""
 
       # de_DE.UTF-8 --> de_DE
       dot_pos = Builtins.search(@language_long, ".")
```
Comment (lslezak) [inline src/lib/y2packager/clients/inst_productsources.rb]: @mvidner recently created a class that handled locale names, maybe we should use it here...

File: src/lib/y2packager/resolvable.rb [hunk 5]
```diff
@@ -91,18 +91,28 @@ def initialize(hash)
...
+      load_attribute(method)
+      super unless instance_variable_defined?("@#{method}")
+      raise ArgumentError, "Method #{method} does not accept arguments" unless args.empty?
+
+      instance_variable_get("@#{method}")
+    end
+
+    # defines for dynamic methods also respond_to?
+    def respond_to_missing?(method, _private)
+      return true if instance_variable_defined?("@#{method}")
```
Comment (lslezak) [inline src/lib/y2packager/resolvable.rb]: This whole function body can be replaced by
```ruby
instance_variable_defined?("@#{method}") || UNIQUE_ATTRIBUTES.include?(method.to_sym)
```

File: src/modules/PackageSlideShow.rb [hunk 6]
```diff
@@ -89,7 +88,7 @@ def ResetPackageSummary
       @removed_packages = 0
       @total_downloaded = 0
       @total_installed = 0
-      #	avg_download_rate = 0;
+      #  avg_download_rate = 0;
```
Comment (lslezak) [inline src/modules/PackageSlideShow.rb]: We should just delete this dead code...

File: src/modules/ProductLicense.rb [hunk 7]
```diff
@@ -348,9 +347,7 @@ def AskLicensesAgreementWithHeading(dirs, _patterns, action, enable_back,
         tmp_licenses = tmp_licenses_ref.value
         available_langs = available_langs_ref.value
         license_ident = license_ident_ref.value
-        if !license_ident.nil?
-          license_idents = Builtins.add(license_idents, license_ident)
-        end
+        license_idents = Builtins.add(license_idents, license_ident) if !license_ident.nil?
```
Comment (lslezak) [inline src/modules/ProductLicense.rb]:
```suggestion
        license_idents << license_ident if license_ident
```

File: src/modules/SourceManagerSLP.rb [hunk 8]
```diff
@@ -24,9 +23,7 @@ def main
 
     def Icon(icon_name)
       ui_info = UI.GetDisplayInfo
-      if Ops.get_boolean(ui_info, "HasLocalImageSupport", false) == false
-        return Empty()
-      end
+      return Empty() if Ops.get_boolean(ui_info, "HasLocalImageSupport", false) == false
```
Comment (lslezak) [inline src/modules/SourceManagerSLP.rb]:
```suggestion
      return Empty() if !Ops.get_boolean(ui_info, "HasLocalImageSupport", false)
```

File: src/include/packager/repositories_include.rb (hunk 1 above)
Comment (jreidinger) [reply src/include/packager/repositories_include.rb]: I did it intentionally. I prefer simple exit criteria. For me it is two logical statement.

1. not samba
2. already installed

File: src/lib/packager/clients/repositories.rb (hunk 3 above)
Comment (jreidinger) [reply src/lib/packager/clients/repositories.rb]: btw is it not a typo that "cd:" and "dvd" without colon?

File: src/lib/y2packager/clients/inst_productsources.rb (hunk 4 above)
Comment (jreidinger) [reply src/lib/y2packager/clients/inst_productsources.rb]: yep, but not goal of this PR.

File: src/lib/y2packager/resolvable.rb (hunk 5 above)
Comment (jreidinger) [reply src/lib/y2packager/resolvable.rb]: yes, for me the one with early exit is more readable as it follows structure of method_missing, but I can change it.

File: src/modules/PackageSlideShow.rb (hunk 6 above)
Comment (jreidinger) [reply src/modules/PackageSlideShow.rb]: :+1: this was auto correction

File: src/lib/packager/clients/repositories.rb (hunk 3 above)
Comment (lslezak) [reply src/lib/packager/clients/repositories.rb]: This is quite tricky, it's the result of
```ruby
schema = Builtins.tolower(Builtins.substring(url2, 0, 3))
```

So for "dvd:///" it is just "dvd", but for "cd:///" it is "cd:" including the colon... :anguished:

File: src/lib/y2packager/resolvable.rb (hunk 5 above)
Comment (lslezak) [reply src/lib/y2packager/resolvable.rb]: ok, I see

File: src/include/packager/repositories_include.rb (hunk 1 above)
Comment (lslezak) [reply src/include/packager/repositories_include.rb]: ok

File: src/lib/y2packager/solvable_pool.rb [hunk 9]
```diff
@@ -27,12 +27,11 @@ def initialize
...
-      gz = open(primary_xml)
-      fd = Solv.xfopen_fd(primary_xml, gz.fileno)
-      repo.add_rpmmd(fd, nil, 0)
-      pool.createwhatprovides
-    ensure
-      gz.close if gz
+      File.open(primary_xml) do |gz|
+        fd = Solv.xfopen_fd(primary_xml, gz.fileno)
+        repo.add_rpmmd(fd, nil, 0)
+        pool.createwhatprovides
```
Comment (lslezak) [inline src/lib/y2packager/solvable_pool.rb]: NP: I'd just move this line outside the block, it's not related to the opened file.

I completely forgot that I could use the block here...

**Pull request merged**